### PR TITLE
Test on Python 3.13

### PR DESCRIPTION
**Pull request by DimitriPapadopoulos**
Python 3.13 is not available yet in some runners, use Python 3.12 instead.

### CODE REVIEW

**Comment by matthew-brett** [comment]
It looks like Python 3.13 is not yet available on Ubuntu. 

```
Installed versions
  Version 3.13 was not found in the local cache
  Error: The version '3.13' with architecture 'x64' was not found for Ubuntu 22.04.
  The list of all available versions can be found here: https://raw.githubusercontent.com/actions/python-versions/main/versions-manifest.json
```

 Let's stick to Python 3.12 as the default version to test on, for now, but test on Python 3.13 for the platforms on which it is currently available.   How are we for wheels on Python 3.13?

**Comment by DimitriPapadopoulos** [comment]
Python 3.13 will be released on 7 October. I suspect it will be available for all tests soon after that.

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -23,7 +23,7 @@ jobs:
   test:
     strategy:
       matrix:
-        python_version: ["3.8", "3.9", "3.10", "3.11", "3.12"]
+        python_version: ["3.8", "3.9", "3.10", "3.11", "3.12", "3.13"]
```
Comment (effigies) [inline .github/workflows/test.yml:26]:
```suggestion
        python_version: ["3.8", "3.9", "3.10", "3.11", "3.12", "3.13-dev"]
```

File: .github/workflows/doc-build.yml [hunk 2]
```diff
@@ -15,7 +15,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        python-version: ["3.12"]
+        python-version: ["3.13"]
```
Comment (effigies) [inline .github/workflows/doc-build.yml:18]:
```suggestion
        python-version: ["3"]
```

File: .github/workflows/coverage.yml [hunk 3]
```diff
@@ -15,7 +15,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        python-version: ["3.12"]
+        python-version: ["3.13"]
```
Comment (effigies) [inline .github/workflows/coverage.yml:18]:
```suggestion
        python-version: ["3"]
```

**Comment by effigies** [comment]
Oh I misunderstood and thought 3.13 wasn't installing.

File: .github/workflows/test.yml (hunk 1 above)
Comment (matthew-brett) [reply .github/workflows/test.yml:26]: Yes - this does not appear to be necessary ...

**Comment by matthew-brett** [comment]
Thanks @DimitriPapadopoulos

**Pull request merged**